src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jpa/entity/DynamicFrameworkContract.kt

```diff
@@ -21,11 +22,12 @@ data class DynamicFrameworkContract(
   @NotNull val startDate: LocalDate,
   @NotNull val endDate: LocalDate,
 
-  @ManyToOne @JoinColumn(name = "nps_region_id")
-  val npsRegion: NPSRegion? = null,
-
-  @ManyToOne @JoinColumn(name = "pcc_region_id")
-  val pccRegion: PCCRegion? = null,
+  // a contract is offered for a single NPS region,
+  // or one-or-more PCC regions within an NPS region.
+  // if `npsRegion` is set, `pccRegions` should contain
```

Comment:
Can we make the model more explicit in that we make both fields mandatory, so that a npsRegion always applies, and the pccRegion list contains all or some of the pccRegions. Where intervention applies to all pccRegions within the npsRegion, then all are present, otherwise only the subset that applies are present? Apologies if I've missed a constraint.